src/tools/idlc/src/descriptor.c

```diff
@@ -1297,6 +1297,22 @@ static int print_flags(FILE *fp, struct descriptor *descriptor)
   if (descriptor->flags & DDS_TOPIC_FIXED_KEY)
     vec[len++] = "DDS_TOPIC_FIXED_KEY";
 
+  bool fixed_size = true;
+  for (uint32_t op = 0; op < descriptor->instructions.count && fixed_size; op++)
+  {
+    struct instruction i = descriptor->instructions.table[op];
+    if (i.type != OPCODE)
+      continue;
+
+    if (((i.data.opcode.code>>16)&0xFF) == DDS_OP_VAL_STR ||
+      ((i.data.opcode.code >> 16) & 0xFF) == DDS_OP_VAL_BST ||
+      ((i.data.opcode.code >> 16) & 0xFF) == DDS_OP_VAL_SEQ)
+      fixed_size = false;
```

Comment:
A few points:
* Shouldn't this also recurse into structs & unions?
* I also suspect there are functions or macros to extract the opcode (there are definitions in `ddsi_cdrstream.h`, but I'm guessing the IDL compiler already has some)
* The `flags` field was inherited, but some things were missing in the old IDL compiler and some fields were computed incorrectly. Instead of fixing that old piece of junk, I instead started computing (some of) them while constructing the `ddsi_sertype_default`, e.g., `dds_stream_check_optimize` & `dds_stream_countops` in `ddsi_cdrstream.c`. Doing it twice is a bit silly, but a run-time verification really is required if the types can ever come from an untrusted source (which is definitely possible nowadays with the type lookup capability). So I would like to consider doing it in `ddsi_cdrstream.c` instead.

Reply:
I thought that the current iteration of the opcode generation fully unrolls all structures, so that by walking over the complete list of opcodes of a structure you would be able to see "into" all present structs and unions as well, but I will examine this further.
I will look into ddsi_cdrstream.c and see what kind of inspiration I am able to glean from there.

Reply:
As far as I can see, all structures are fully unrolled in the opcodes